nemo/collections/asr/data/audio_to_text.py

```diff
@@ -1444,3 +1445,55 @@ def __call__(self, text):
             world_size=world_size,
             return_sample_id=return_sample_id,
         )
+
+
+class BucketingDataset(IterableDataset):
+    """
+    lodfehlkfeh fqpfjqfpopf fpqojf qpoj!!!
+    """
+
+    def __init__(
+        self, dataset: IterableDataset, bucketing_batch_size: int,
+    ):
+        self.wrapped_dataset = dataset
+        self.bucketing_batch_size = bucketing_batch_size
+        super().__init__()
+
+    def _collate_fn(self, batch):
+        return _speech_collate_fn(batch[0], self.wrapped_dataset.pad_id)
+
+    def __iter__(self):
+        return BucketingIterator(
+            wrapped_iter=self.wrapped_dataset._dataset.__iter__(), bucketing_batch_size=self.bucketing_batch_size
+        ).__iter__()
+
+    def __len__(self):
+        # print(
+        #     f"dataset size={int(math.ceil(len(self.wrapped_dataset.collection)/float(self.bucketing_batch_size)))}, bucketing_batch_size={self.bucketing_batch_size}"
+        # )
+        return int(math.ceil(len(self.wrapped_dataset.collection) / float(self.bucketing_batch_size)))
+
+
+class BucketingIterator:
+    def __init__(self, wrapped_iter, bucketing_batch_size):
+        self.wrapped_iter = wrapped_iter
+        self.bucketing_batch_size = bucketing_batch_size
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        batches = []
+        for idx in range(self.bucketing_batch_size):
+            try:
+                sample = next(self.wrapped_iter)
+            except StopIteration:
+                break
+            except Exception as e:
+                logging.error(e)
```

Comment:
Why not raise the error instead of just log it? Below it will append None then into batches.append(..)

Reply:
dropped it.